core/src/main/java/com/linecorp/armeria/common/stream/StreamMessageInputStream.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.stream;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.function.Function;
+
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.internal.common.stream.StreamMessageUtil;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+
+final class StreamMessageInputStream<T> extends InputStream {
+
+    public static <T> StreamMessageInputStream<T> of(StreamMessage<T> source,
+                                                     Function<T, Byte> byteConverter) {
+        return new StreamMessageInputStream<>(source, byteConverter);
+    }
+
+    private final ByteBuf buf = ByteBufAllocator.DEFAULT.buffer();
```

Comment:
I'm not sure if I'm implementing `StreamMessageInputStream` in right way 😅 
(So I add `Draft` on title of this PR)

Based on this [slack](https://line-armeria.slack.com/archives/C1NGPBUH2/p1643038254042400) discussion, I try to implement this `StreamMessageInputStream` in reactive way.

I mostly refered armeria#[ByteBuffersBackedInputStream](https://github.com/line/armeria/blob/master/resteasy/src/main/java/com/linecorp/armeria/internal/common/resteasy/ByteBuffersBackedInputStream.java).java class(I think this class is similar to what I want), use `ByteBuf` as buffer to store bytes that emitted from `StreamMessage`.

If you have any suggestion, feel free to tell to me!

Reply:
![image](https://user-images.githubusercontent.com/34854527/152333037-e8cb14f1-9d8a-42c9-82b2-312ef7e9e054.png)

(What I intended)

Reply:
If the `source` publishes `HttpData` mentioned in https://github.com/line/armeria/pull/4059#discussion_r802364071, I don't think we need to create a temporarily `ByteBuf`.

Reply:
✅ I removed temporary `ByteBuf`